`CLANG_ASSISTED_PARSING` is misprocessing `.h` files

Setup the following Doxyfile configuration:
    
    WARN_AS_ERROR          = YES
    CLANG_ASSISTED_PARSING = YES
    CLANG_OPTIONS +=-I/usr/local/lib/clang/18/include

Write the following program:

    /// The main header file
    /// 
    /// @file main.h
    #pragma once

    /// An enum
    enum blob_e {
       FIRST = 0x01,  ///< The first thing
       SECOND = 0x02  ///< The second thing
    };

and

    /// The main file
    /// 
    /// @file main.c

    #include <stdio.h>

    #include "main.h"

    /// Program entry point
    ///
    /// Reference to #FIRST
    ///
    /// @return The program's result code
    int main() {
       printf( "Hello world\n" );
    }

This generates the following error:

    error: main.h:4:9: warning: #pragma once in main file [-Wpragma-once-outside-header] [clang]

If you disable `CLANG_ASSISTED_PARSING` then the error goes away.

------

Here's another usecase that's broken...

Move `main.h` into `./include`:
- `./main.c`
- `./include/main.h` 

Naturally, Doxygen will complain that `main.c` can't find `main.h`, so we do this:
`CLANG_OPTIONS +=-I./include`

...but now Doxygen throws the following error:

    main.c:12: error: explicit link request to 'FIRST' could not be resolved 

...because Doxygen is not processing the `.h` file.

Adding `INPUT += ./include` to the Doxyfile
...gets us back to `main.h:4:9: warning: #pragma once in main file`


**Expected behavior**
Doxygen w/ `CLANG_ASSISTED_PARSING` should be able to find & process header files

**Version**
    $ doxygen --version
    1.10.0 (GIT-NOTFOUND)

--- a/src/clangparser.cpp
+++ b/src/clangparser.cpp
@@ -180,9 +180,9 @@ void ClangTUParser::parse()
     // we use the source file to detected the language. Detection will fail if you
     // pass a bunch of .h files containing ObjC code, and no sources :-(
     SrcLangExt lang = getLanguageFromFileName(fileName);
+    QCString fn = fileName.lower();
     if (lang==SrcLangExt::ObjC || p->detectedLang!=DetectedLang::Cpp)
     {
-      QCString fn = fileName.lower();
       if (p->detectedLang!=DetectedLang::Cpp &&
           (fn.endsWith(".cpp") || fn.endsWith(".cxx") ||
            fn.endsWith(".cc")  || fn.endsWith(".c")))
@@ -200,7 +200,13 @@ void ClangTUParser::parse()
     }
     switch (p->detectedLang)
     {
-      case DetectedLang::Cpp:    argv.push_back(qstrdup("c++"));           break;
+      case DetectedLang::Cpp:
+        if (fn.endsWith(".hpp") || fn.endsWith(".hxx") ||
+            fn.endsWith(".hh")  || fn.endsWith(".h"))
+          argv.push_back(qstrdup("c++-header"));
+        else
+          argv.push_back(qstrdup("c++"));
+        break;
       case DetectedLang::ObjC:   argv.push_back(qstrdup("objective-c"));   break;
       case DetectedLang::ObjCpp: argv.push_back(qstrdup("objective-c++")); break;
     }

